version_test.go

```diff
@@ -116,6 +116,68 @@ func TestK0sComparison(t *testing.T) {
 	False(t, b.Equal(a))
 }
 
+func TestVersion_TotalOrder(t *testing.T) {
+	totalOrder := [...]*version.Version{
+		version.MustParse("0.9.9"),
+		version.MustParse("1.0-a"),
+		version.MustParse("1.0.0-a"),
+		version.MustParse("1"),
+		version.MustParse("1.0"),
+		version.MustParse("1.0.0"),
+		version.MustParse("1.0.1"),
+		version.MustParse("1.0.1+k0s.10"),
+		version.MustParse("1.0.9+k0s.255"),
+		version.MustParse("1.1.0-alpha.1+k0s.2"),
+		version.MustParse("1.1.0-beta.1+k0s.1"),
+		version.MustParse("1.1.0-rc.1+k0s.0"),
+		version.MustParse("1.1.0+k0s.0"),
+		version.MustParse("1.1.0+k0s.1"),
+		version.MustParse("1.1.0+k0s.2"),
+		version.MustParse("1.1.1-rc.1+k0s.0"),
+		version.MustParse("1.1.1+k0s.0"),
+		version.MustParse("2-a"),
+		version.MustParse("2.0-a"),
+		version.MustParse("2.0.0-a"),
+		version.MustParse("2"),
```

Comment:
I think this list might use some restructuring or cleaning up. Are all the cases unique in what they are demonstrating? Maybe break it into smaller pieces? 

I mean at least something like:

```go
func TestVersion_TotalOrder(t *testing.T) {
	totalOrder := [...]*version.Version{
		version.MustParse("0.9.9"),
		// 1-3 digit v1 prerelease should be greater than 0.9 stable
		version.MustParse("1-a"), 
		version.MustParse("1.0-a"), 
		version.MustParse("1.0.0-a"),
		// 1-3 digit stable should be greater than its prerelease
		version.MustParse("1"),
		version.MustParse("1.0"),
		version.MustParse("1.0.0"),
...
...
```

Reply:
> I think this list might use some restructuring or cleaning up. Are all the cases unique in what they are demonstrating?

The initial version of this list was constructed based on all the version strings used in the existing tests. Then I added a few where I thought would be holes in the list.

> Maybe break it into smaller pieces?

I'll add comments. Let's see if we can boil this down a bit.

Reply:
I'm using a different approach right now, by auto-generating the test cases.